### PR TITLE
Reduce the application names list in the stats API

### PR DESCRIPTION
**Pull request by ewels**
This code is horrible. I expect Denis will want to rewrite it all and that's fine. If not, it works the way I want it to at least.

### CODE REVIEW

File: status/statistics.py [hunk 1]
```diff
@@ -22,19 +23,55 @@ def get_stats_data(db,view,gl=None):
...
+    for year in raw_data:
+        data[year] = {}
+        for c in raw_data[year]:
+            try:
+                data[year][app_cats[c]] += raw_data[year][c]
+            except KeyError:
+                try:
+                    data[year][app_cats[c]] = raw_data[year][c]
+                except KeyError:
+                    data[year][c] = raw_data[year][c]
```
Comment (senthil10) [inline status/statistics.py]: If you want to reduce somelines (or hurt someones head), you could replace `try` and `except` loop with

```
data[year][app_cats.get(c, c)] = data[year].get(app_cats.get(c, c),"") + raw_data[year][c]
```

Comment (ewels) [inline status/statistics.py]: Aww, @senthil10 - that's far too clean and tidy! I had fun putting in this PR as I knew it was horrible and would offend everyone's brains.

Your suggestion is most elegant. I shall adopt it.

File: status/statistics.py [hunk 2]
```diff
@@ -22,19 +23,55 @@ def get_stats_data(db,view,gl=None):
...
+    return data
+
+def clean_application_keys(raw_data):
```
Comment (ewels) [inline status/statistics.py]: @senthil10 - can you think of a nice way to combine these two functions whilst you're at it?

Comment (senthil10) [inline status/statistics.py]: @ewels : How about this one ?

```
def clean_application_keys(raw_data, year=False):
    with open('application_keys.yaml') as f:
        app_cats=yaml.load(f)
    data = {}
    if year:
        for year in raw_data:
            data[year] = clean_application_keys(raw_data[year])
    for c in raw_data:
        data[app_cats.get(c, c)] = data.get(app_cats.get(c, c),"") + raw_data[c]
    return data
```

not sure of it though 😅

Comment (ewels) [inline status/statistics.py]: Yeah I thought of that. But then I have to call the function in two different ways which I'd rather avoid. I did something similar but using an `assert` statement to figure it out automatically in cd0a37c8770bc753d813ec5cabd978828e171866

Comment (ewels) [inline status/statistics.py]: [Here's the line I'm talking about](https://github.com/SciLifeLab/genomics-status/pull/314/commits/cd0a37c8770bc753d813ec5cabd978828e171866#diff-7c8921bccc4452bce88f8971e9d55163R34)

File: status/statistics.py [hunk 3]
```diff
@@ -22,58 +23,68 @@ def get_stats_data(db,view,gl=None):
...
+    try:
+        for year in raw_data:
+            assert int(year) > 2000
+            data[year] = {}
+            for c in raw_data[year]:
+                data[year][app_cats.get(c, c)] = data[year].get(app_cats.get(c, c),0) + raw_data[year][c]
+    except (ValueError, AssertionError):
+        for c in raw_data:
+            data[app_cats.get(c, c)] = data.get(app_cats.get(c, c),0) + raw_data[c]
+    return data
```
Comment (senthil10) [inline status/statistics.py]: @ewels : Nice 😄  👍

**Comment by ewels** [comment]
Closing because Denis wrote #315

**Pull request closed without merging**